src/CoreWCF.Primitives/tests/ServiceKnownTypeTests.cs

```diff
@@ -0,0 +1,223 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.Serialization;
+using DispatcherClient;
+using Helpers;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace CoreWCF.Primitives.Tests
+{
+    public class ServiceKnownTypeTests
+    {
+        internal static System.ServiceModel.ChannelFactory<TContract> CreateFactory<TContract, TService>() where TService : class
+        {
+            System.ServiceModel.ChannelFactory<TContract> factory = DispatcherHelper.CreateChannelFactory<TService, TContract>(
+                (services) =>
+                {
+                    services.AddScoped<TService>();
+                });
+            return factory;
+        }
+
+        internal static TContract CreateChannel<TContract>(System.ServiceModel.ChannelFactory<TContract> factory)
+        {
+            factory.Open();
+            TContract channel = factory.CreateChannel();
+            ((System.ServiceModel.Channels.IChannel)channel).Open();
+            return channel;
+        }
+
+        internal static void CloseChannel<TContract>(System.ServiceModel.ChannelFactory<TContract> factory, TContract channel)
+        {
+            ((System.ServiceModel.Channels.IChannel)channel).Close();
+            factory.Close();
+            TestHelper.CloseServiceModelObjects((System.ServiceModel.Channels.IChannel)channel, factory);
+        }
+
+        [Fact]
+        public static void ServiceKnownTypeException()
+        {
+            var factory = ServiceKnownTypeTests.CreateFactory<IPingServiceEx, PingServiceEx>();
+            var channel = ServiceKnownTypeTests.CreateChannel(factory);
+            Assert.Throws<System.Xml.XmlException>(() =>
+            {
+                BaseMsg msg = channel.Ping("hello");
+            });
+            ServiceKnownTypeTests.CloseChannel(factory, channel);
+        }
+
+        [Fact]
+        public static void ServiceKnownTypeTest1()
+        {
```

Comment:
These test names are a bit obscure. Can you rename them to have a bit of description. E.g. this one could be `ServiceKnownTypeInlineType` as the type is inline to the attribute. Maybe add a comment at the start with the `ServiceKnownType` usage. So something like this:
```c#
        public static void ServiceKnownTypeInlineType()
        {
            // Testing [ServiceKnownType(typeof(DerivedMsg))]
            /* Rest of test */
        }
```

Reply:
Done